### DISCUSSION TITLE
Other UPDI Adapters :-)

### DISCUSSION THREAD

**Discussion by HolleL** [SpenceKonde/megaTinyCore#429] [Show and tell]
Hello (sorry for my bad english)

I have seen the new Variants of Adapters (prog.py), there are very similar to some typical Update-Adapters for RC-modelling, so i have just test it with a Attiny412. (Serial-UPDI Slow).

-With a Graupner/Hott and Jeti Adapter(One-Wire) i can upload without Resistor. 

-With a BLHeli ESC Adapter run without and a 1K5 resistor, but need a ~10µF Cap on Reset. 
The BLHeliadapter is for programming the  OpenSource  ESCs on a Arduino. 
https://github.com/bitdump/BLHeli

BLHeli have a One-Wire Bootloader based on the great AVROOTLOADER from Hagen https://github.com/damadmai/AVRootloader. 
I use the AVRooloader for the most of my Attiny Projekts (old Attiny and Atmega) in RCmodelling, its simple to flash over the Servo-cable. (i  missed this a little, but the new attinys are so great ;-) 

Now I can use my old Adapters for the new Attinys, many thanks, Spence :-) 

Have a nice day, Holger

**Comment by SpenceKonde** [comment]
Glad to hear you're finding it useful. 

Surprised to hear of a need for a "reset-disable" cap - normally autoreset is mediated by the RTS and DTR pins. SerialUPDI should not be changing the state of those pins. (If it is, that is a bug that needs to be corrected.) Besides that, the only way I can imagine that being the case is if there was also a microcontroller on the thing with the serial adapter which didn't use serial except for the bootloader, and so the reset had to be disabled to leave it in it's normal, non-serial-using mode.

Those programmers almost certainly have some resistance in series with their TX pin already (doesn't work without it), 1k ~ 2.2k is almost ubiquitous here. In fact, I only know of two serial adapter board designs that don't have a built-in TX resistor - one of them is the first PCB I ever designed, and the other one is... well, judging by the layout, the first PCB that someone else designed, only unlike me, they've been getting hundreds of them made for years and are selling them. They had more good ideas than I did on my first CH340 board (in fact, my original design didn't really have any of those),  but they made many, many stupid mistakes. Don't buy those little green CH340G serial adapters with the voltage switch and micro USB port, heh...

**Reply by hmeijdam** [reply]
Are you referring to this little programmer as a no-go? It has no TX resistor and after soldering only one diode on the back it's working actually very well at 460800 baud. And I did not have to remove the TX/RX leds
![Serial programmer](https://user-images.githubusercontent.com/9595465/119570544-8ea4f080-bdb0-11eb-9415-dc1af1ebfd25.jpg)
